cg/meta/upload/fohm/fohm.py

```diff
@@ -116,16 +90,64 @@ def daily_pangolin_list(self) -> list[Path]:
         if not self._daily_pangolin_list:
             for case_id in self._cases_to_aggregate:
                 self._daily_pangolin_list.append(
-                    self.housekeeper_api.get_file_from_latest_version(
-                        bundle_name=case_id, tags=["pangolin-typing-fohm"]
-                    ).full_path
+                    Path(
+                        self.housekeeper_api.get_file_from_latest_version(
+                            bundle_name=case_id, tags={FohmTag.PANGOLIN_TYPING}
+                        ).full_path
+                    )
                 )
         return self._daily_pangolin_list
 
     @property
     def dry_run(self):
         return self._dry_run
 
+    @staticmethod
+    def get_reports_contents(file_paths: list[Path]) -> list[dict]:
+        """Return a list of dicts with all CSV files reports."""
+        csv_reports = [
+            ReadFile.get_content_from_file(
+                file_format=FileFormat.CSV, file_path=file_path, read_to_dict=True
+            )
+            for file_path in file_paths
+        ]
+        reports: list[dict] = []
+        for report in csv_reports:
+            reports.extend(iter(report))
+        return reports
+
+    @staticmethod
+    def validate_fohm_complementary_reports(reports: list[dict]) -> list[FohmComplementaryReport]:
+        """Validate FOHM complementary reports."""
+        complementary_reports: list[FohmComplementaryReport] = []
+        for report in reports:
+            complementary_report = FohmComplementaryReport.model_validate(report)
+            complementary_reports.append(complementary_report)
+        return complementary_reports
+
+    @staticmethod
+    def validate_fohm_pangolin_reports(reports: list[dict]) -> list[FohmPangolinReport]:
+        """Validate FOHM Pangolin reports."""
+        pangolin_reports: list[FohmPangolinReport] = []
+        for report in reports:
+            pangolin_report = FohmPangolinReport.model_validate(report)
+            pangolin_reports.append(pangolin_report)
+        return pangolin_reports
+
+    @staticmethod
+    def get_sars_cov_complementary_reports(
+        reports: list[FohmComplementaryReport],
+    ) -> list[FohmComplementaryReport]:
+        """Return all "Sars-cov2" reports from reports."""
+        return [report for report in reports if re.search(SARS_COV_REGEX, report.sample_number)]
```

Comment:
I don't understand what this one does. What does it remove?

Reply:
It rather returns reports for only samples matching the regexp which is specific for sars-cov.